src/boundary/boundary_p.cc

```diff
@@ -318,6 +411,46 @@ boundary_p( particle_bc_t       * RESTRICT pbc_list,
           }
           #endif
 
+          #ifdef VPIC_PARTICLE_ANNOTATION
+          int n_send_face = n_send[ face ] - 1; // Minus one because it was already incremented above
+          float* RESTRICT ALIGNED(128) p_annotation = (float*) &(cab[face][n_send_face * max_cas]);
+
+          //printf("<%d> %dth particle of species %s has %d annotation that need to go into the buffer starting at %p\n", rank(), i, sp->name, sp_has_annotation, p_annotation);
+
+          // Store species ID
+          {
+            // Oh god scary...
+            int* p_annotation_int = (int*) p_annotation;
```

Comment:
I'm overall a bit bamboozled by what this is doing.

We load the pointer, we set what it's pointing to, then we nudge the pointer along and store it again to use below?

It looks like what we actually want to do is... store `sp_id` into the `cab` using `p_annotation` to do that? But I feel like I'm missing something

Reply:
Luckily this part of the code is gone now.

Reply:
Is this gone now? it doesn't look gone

Reply:
Per slack, I think we should think about this a bit more?

Reply:
Probably. But we need to combine a 64 bit particle ID with a variable number of 32 bit float annotations into a buffer that is save to send over MPI. Saying "I just count bytes for size and make the types fit" is the best I could come up with. And despite the "Oh god scary" because we lose type safety, this is not one of the spots where I had bugs during development.